src/util/converter/factory.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA ;
+ */
+
+import _ from 'lodash';
+
+/**
+ * Defines a converter processor runtime.
+ * @callback converterProcessorRuntime
+ * @param {string} text - The text to convert.
+ * @param {object} [config] - An optional config object that may contain processor specific configuration.
+ * @returns {string} - Returns the converted text.
+ */
+
+/**
+ * Defines a converter processor.
+ * @typedef {object} converterProcessor
+ * @property {string} name - The name of the converter processor. It needs to be unique within the collection.
+ * @property {converterProcessorRuntime} convert - The processor runtime that will actually convert the text.
+ */
+
+/**
+ * Checks a converter processor is valid.
+ * @param {converterProcessor} processor - The converter processor to validate.
+ * @returns {boolean} - Returns `true` if the converter processor is valid ; returns `false` otherwise.
+ */
+function validateProcessor(processor) {
+    if (!_.isPlainObject(processor)) {
```

Comment:
What's the purpose of this check? And by the way, this is the only place `lodash` is used here

Reply:
Indeed, this is the only reason for `lodash` to be there. The `isPlainObject()` could be inlined, but the mechanism is complex and I found it better to use lodash. But, yes, we could only rely on a `typeof object` check.